lib/interface/cli/helpers/general.js

```diff
@@ -1,15 +1,43 @@
+const debug = require('debug')('codefresh:cli:helpers:general');
 const Promise = require('bluebird');
 const _ = require('lodash');
 const fs = require('fs');
 const yaml = require('js-yaml');
-const defaults = require('../defaults');
 const CFError = require('cf-errors');
 const path = require('path');
+const semver = require('semver');
+const request = require('requestretry');
+const defaults = require('../defaults');
 const Output = require('../../../output/Output');
 const { sdk } = require('../../../logic');
 
 const isDebug = () => (process.env.DEBUG || '').includes(defaults.DEBUG_PATTERN);
 
+async function getApiVersion(token, url) {
+    const RequestOptions = {
+        url: `${url}/api/user`,
+        headers: {
+            Authorization: token,
+        },
+        json: true,
+        maxAttempts: 3, // Retry up to 3 times
+        timeout: 10000,
+    };
+    try {
+        const res = await request(RequestOptions);
+        return _.get(res, 'headers.api_version');
+    } catch (error) {
+        debug('Failed to get API version:', error);
+        return null;
+    }
+}
+
+async function isCompatibleApiVersion({ supportedVersion }) {
+    const token = _.get(sdk, 'config.context.token');
+    const url = _.get(sdk, 'config.context.url', defaults.URL);
+    const apiVersion = token ? await getApiVersion(token, url) : null;
+    return apiVersion && semver.gte(apiVersion, supportedVersion);
```

Comment:
lets have here a consistent return type - this function should always return boolean and not null/undefined